src/cljs/rems/user_settings.cljs

```diff
@@ -17,9 +25,11 @@
 
 (rf/reg-event-fx
  ::update-user-settings
- (fn [_ [_ settings]]
-   (put! "/api/user-settings"
-         {:params settings
-          :handler (fn [_] (fetch-user-settings!))
-          :error-handler (flash-message/default-error-handler :top "Update user settings")})
-   nil))
+ (fn [{:keys [db]} [_ settings]]
+   (let [user-id (get-in db [:identity :user :eppn])
+         new-settings (merge (:user-settings db) settings)]
+     (when user-id
+       (put! "/api/user-settings"
+             {:params new-settings
+              :error-handler (flash-message/default-error-handler :top "Update user settings")}))
+     {:db (assoc db :user-settings new-settings)})))
```

Comment:
hmm, or should we have separate events for a) updating `:user-settings` in db b) sending the current user settings from the db to the server